From fc37221db5d08eb9553273ad6202a2557cfde131 Mon Sep 17 00:00:00 2001
From: Spyros Trigazis <strigazi@gmail.com>
Date: Fri, 12 Jan 2018 09:56:50 +0000
Subject: [PATCH] Fix comparison of golang versions

Change hack/lib/golang.sh to compare golang
version properly with "sort -s -t. -k 1,1 -k 2,2n -k 3,3n",
which sorts key by key and not as strings.
---
 hack/lib/golang.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hack/lib/golang.sh b/hack/lib/golang.sh
index 3e12b3170a44c..589e965afb42d 100755
--- a/src/k8s.io/kubernetes/hack/lib/golang.sh
+++ b/src/k8s.io/kubernetes/hack/lib/golang.sh
@@ -323,7 +323,7 @@ EOF
   go_version=($(go version))
   local minimum_go_version
   minimum_go_version=go1.9.1
-  if [[ "${go_version[2]}" < "${minimum_go_version}" && "${go_version[2]}" != "devel" ]]; then
+  if [[ "${minimum_go_version}" != $(echo -e "${minimum_go_version}\n${go_version[2]}" | sort -s -t. -k 1,1 -k 2,2n -k 3,3n | head -n1) && "${go_version[2]}" != "devel" ]]; then
     kube::log::usage_from_stdin <<EOF
 Detected go version: ${go_version[*]}.
 Kubernetes requires ${minimum_go_version} or greater.
